src/cegpy/trees/_staged.py

```diff
@@ -623,7 +623,7 @@ def _execute_ahc(
                     if situ_pair[1] in comb:
                         hyperstage_combinations.remove(comb)
 
-        while True:
+        while len(hyperstage_combinations) != 0:
```

Comment:
Can you please give more context to what the bug you encountered was?

Reply:
When the user sets initial_staging = hyperstage, there are no possible stages to merge and so hyperstage_combinations was empty. Previously this would result in an error. With my change the stage merging step is skipped when hyperstage_combinations has length 0.